With:

python: 3.8.x
SQLAlchemy: 1.3.23

```sh
>>> result = db.engine.execute(text(q))
>>> print(type(result))
<class 'sqlalchemy.engine.result.ResultProxy'>

>>> print(result.first())
('xxx', 'Insufficient data', 0)

>>> print(tabulate([row for row in result], headers=result.keys()))
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/tabulate.py", line 1528, in tabulate
    list_of_lists, headers = _normalize_tabular_data(
  File "/usr/local/lib/python3.8/site-packages/tabulate.py", line 1161, in _normalize_tabular_data
    raise ValueError(
ValueError: headers for a list of dicts is not a dict or a keyword

>>> result = db.engine.execute(text(q))
>>> r_tuples = result.fetchall()
>>> print(tabulate([list(r) for r in r_tuples], headers=result.keys()))
relname                    percent_of_times_index_used      rows_in_table
-------------------  -----------------------------  ---------------
xxx                        Insufficient data                            0
...
```

It doesn't seem to suggest that SQLAlchemy is the root cause. I have too many dependencies so I can't try with python 3.6x neither. I'm about to move to 3.10x very soon as well.

2 questions:
- is it possible for you to test this patch on your end?
- are there any intentions to upgrade the dependencies to their latest versions?

--- a/flask_pg_extras/__init__.py
+++ b/flask_pg_extras/__init__.py
@@ -30,6 +30,8 @@ def db_execute_results(db, q):
     """
     result = db.engine.execute(text(q))
 
-    print(tabulate([row for row in result], headers=result.keys()))
+    result_fetchall = result.fetchall()
+
+    print(tabulate([list(r) for r in result_fetchall], headers=result.keys()))
 
     return result

